composables/drop/useGenerativeDropMint.ts

```diff
@@ -28,47 +26,38 @@ export type DropCollectionById = {
   }
 }
 
-function useCollectionData(collectionId, client) {
-  const { accountId } = useAuth()
-  const { vueApp } = useNuxtApp()
-  return vueApp.runWithContext(() =>
-    useQuery<DropCollectionById | null>({
-      queryKey: ['collection-drop-data', client, collectionId, accountId],
-      queryFn: () =>
-        collectionId.value
-          ? useAsyncQuery<DropCollectionById | null>({
-            clientId: client.value,
-            query: unlockableCollectionById,
-            variables: {
-              id: collectionId.value,
-              search: { issuer_eq: accountId.value },
-            },
-          }).then(res => res.data.value)
-          : null,
-    }),
-  )
-}
-
 export function useCollectionEntity() {
   const { drop } = useDrop()
-  const { client } = usePrefix()
-  const { data: collectionData } = useCollectionData(
-    computed(() => drop.value?.collection),
-    client,
-  )
-  const maxCount = computed(() => collectionData.value?.collectionEntity?.max)
-  const description = computed(
-    () => collectionData.value?.collectionEntity?.meta?.description ?? '',
-  )
-  const collectionName = computed(
-    () => collectionData.value?.collectionEntity?.name ?? '',
-  )
-  const nftCount = computed(
-    () => collectionData.value?.collectionEntity?.nftCount ?? 0,
-  )
+  const maxSupply = ref()
+  const nftCount = ref()
+  const description = ref()
+  const collectionName = ref()
+
+  watchEffect(async () => {
+    if (drop.value.collection) {
+      const api = await useApi().apiInstance.value
+      const [queryCollectionConfig, queryCollection, queryCollectionMetadata] = await Promise.all([
+        api.query.nfts.collectionConfigOf(drop.value.collection),
+        api.query.nfts.collection(drop.value.collection),
+        api.query.nfts.collectionMetadataOf(drop.value.collection),
+      ])
```

Comment:
How about the drops on base (EVM) network? IMO it does not work with the RPC interface.

Reply:
good question 👍🏻
it also works on evm, let me show it to you

Reply:
- https://ogi-feat--evm-api-for-item-d.workers-ogi.pages.dev/api/evm/item/0xa0e2bca2145b692670125dab579d5075f6a10473-16
- https://ogi-feat--evm-api-for-item-d.workers-ogi.pages.dev/api/evm/item/0x61fa91bc4df01c199f2c9576a2fe10ea206a55e1-33

code: https://github.com/kodadot/workers/pull/324/files

Reply:
However, it does not work when I open the drop page on base, due to the API endpoint.
![image](https://github.com/user-attachments/assets/bc5cc35d-63ce-4cc8-981b-41522467f497)


https://deploy-preview-10713--koda-canary.netlify.app/base/drops/quadz


> code: https://github.com/kodadot/workers/pull/324/files

It fetches the data directly from the contract on chain instead of API though 👀

Reply:
Aah, I see. Sorry, I misunderstood your question before. Noted, I will add that also for evm based, either using sdk or api on ogi. then adjust it to something like this:

```
if (drop.value.collection) {
	if (substrate) {}
	if (evm) {}
}
```

Reply:
wdyt?

Reply:
> Aah, I see. Sorry, I misunderstood your question before. Noted, I will add that also for evm based, either using sdk or api on ogi. then adjust it to something like this:
> 
> ```
> if (drop.value.collection) {
> 	if (substrate) {}
> 	if (evm) {}
> }
> ```

Yeah that's a good idea 👍

Reply:
updated, requires this deployment: https://github.com/kodadot/workers/pull/324

Reply:
Nice! Let me know when I can test it.